codec-http/src/main/java/io/netty/handler/codec/http/cookie/ClientCookieEncoder.java

```diff
@@ -91,12 +121,66 @@ public String encode(Cookie... cookies) {
         }
 
         StringBuilder buf = stringBuilder();
-        for (Cookie c : cookies) {
-            if (c == null) {
-                break;
+        if (strict) {
+            ArrayList<Cookie> cookiesList = new ArrayList<Cookie>(cookies.length);
```

Comment:
You should be directly using an array here and use Arrays.sort.
This way, you would save an allocation (Collections.sort copies into an internal array and delegate to Arrays.sort).